### ISSUE TITLE
crabserver loosing tasks ?

### ISSUE THREAD

**Issue by ericvaandering** [dmwm/CRAB2#561]
Original Savannah ticket 67735 reported by farinafa on Thu May 20 12:50:28 2010.

Hi,

Seems like the crabserverworker is loosing task after being stuck.

The only log we have is the following:

The task in question which was initialized
samvel_crab_0_100518_070438_h2c3q1
and then the component was stuck (after saving this bit of log), the crabserverworker never took care of the old tasks.

2010-05-18 04:55:33,408:Publishing CrabServerWorkerComponent:CommandArrival
2010-05-18 04:55:33,409:FatWorker worker_8 loading task
2010-05-18 05:05:08,384:Publishing CrabServerWorkerComponent:Submission
2010-05-18 05:05:14,329:FatWorker worker_9 initialized: task samvel_crab_0_100518_070438_h2c3q1
2010-05-18 05:05:14,331:Publishing CrabServerWorkerComponent:CommandArrival
2010-05-18 05:05:14,332:FatWorker worker_9 loading task
2010-05-18 05:05:44,333:Publishing CrabServerWorkerComponent:Submission
2010-05-18 05:05:49,374:FatWorker worker_10 initialized: task samvel_crab_0_100517_173220_59a8ng
2010-05-18 05:05:49,376:FatWorker worker_10 loading task
2010-05-18 05:05:49,376:Publishing CrabServerWorkerComponent:CommandArrival
2010-05-18 05:18:49,329:Publishing CrabServerWorkerComponent:Submission
2010-05-18 05:18:54,367:FatWorker worker_11 initialized: task querten_Mu_RunBelow134987_ok4p09
2010-05-18 05:18:54,368:FatWorker worker_11 loading task
2010-05-18 05:18:54,369:Publishing CrabServerWorkerComponent:CommandArrival
2010-05-18 05:18:54,371:Publishing CrabServerWorkerComponent:Submission
2010-05-18 05:18:59,410:FatWorker worker_12 initialized: task jueugste_crab_0_100518_141759_62pbc0
2010-05-18 05:18:59,412:FatWorker worker_12 loading task
2010-05-18 05:18:59,413:Publishing CrabServerWorkerComponent:CommandArrival
2010-05-18 05:25:19,400:Publishing CrabServerWorkerComponent:Submission
2010-05-18 05:25:24,440:FatWorker worker_13 initialized: task dammann_crab_0_100518_135555_0oqp19
2010-05-18 05:25:24,441:Publishing CrabServerWorkerComponent:CommandArrival
2010-05-18 05:25:24,442:FatWorker worker_13 loading task
2010-05-18 05:26:09,438:Publishing CrabServerWorkerComponent:Submission
2010-05-18 05:26:14,477:FatWorker worker_14 initialized: task jueugste_crab_0_100518_142515_zq5i23
2010-05-18 05:26:14,478:FatWorker worker_14 loading task
2010-05-18 05:26:14,479:Publishing CrabServerWorkerComponent:CommandArrival
2010-05-18 05:32:09,467:Publishing CrabServerWorkerComponent:Submission
2010-05-18 05:32:14,506:FatWorker worker_15 initialized: task jueugste_crab_0_100518_142946_15om0t
2010-05-18 05:32:14,507:Publishing CrabServerWorkerComponent:CommandArrival
2010-05-18 05:32:14,509:FatWorker worker_15 loading task
2010-05-18 07:35:40,926:-----------------------------------------
2010-05-18 07:35:40,927:CrabServerWorkerComponent ver2 Started...
2010-05-18 07:35:40,927:Component dropbox working directory: /var/gftp_cache/CSstoragePath/logs

2010-05-18 07:35:41,461:Resubmission has no more attempts: give up with job robertc_crab_0_100518_004102_3kx2d4_job156
2010-05-18 07:35:41,463:Publishing CrabServerWorkerComponent:SubmitNotSucceeded
2010-05-18 07:35:41,467:Resubmission has no more attempts: give up with job eaguiloc_crab_0_100518_120026_py783h_job42
2010-05-18 07:35:41,469:Publishing CrabServerWorkerComponent:SubmitNotSucceeded
2010-05-18 07:35:41,477:Resubmission has no more attempts: give up with job rcr_135059_19pi8w_job1
2010-05-18 07:35:41,478:Publishing CrabServerWorkerComponent:SubmitNotSucceeded

**Comment by ericvaandering** [comment]
Comment by farinafa on Fri May 21 03:39:46 2010

Ciao Sanjay, 
   if I remind well those tasks were on the cern server. I gave a look at the disaster cache before cleaning it and restarting the database server.

So, I'm almost sure I removed the re-submission messages when I restarted the machinery in order to unlock both TaskTracking and MySQL.

On the other hand, 15 workers are really a large number of thread. I'd suggest to reduce the to 5-6 as for the other servers.

**Comment by ericvaandering** [comment]
Comment by farinafa on Fri May 21 07:04:50 2010

Let me clarify better.

I removed the cache in order to avoid different problems like those already experienced by the cern server and reported in

https://savannah.cern.ch/bugs/index.php?66389

**Comment by ericvaandering** [comment]
Comment by spadhi on Fri May 21 14:47:15 2010

Hi Fabio,

I understand.

However, if the host resources are not loaded, I am not sure about the limitation on threads.

Removeing cache and restarting it can be a temporary measure,
do we understand why this happened in the first place and what
is causing this.

Thanks, Sanjay

**Comment by ericvaandering** [comment]
Comment by farinafa on Thu May 27 04:17:31 2010

Hi!

I think the ticket can be closed. We talked viachat about a good tuning strategy for the threads.

Just one clarification. I removed the cache not to reduce the load on the system but, as the problem looked like some kind of mysql problem too (missing main server process) I preferred to remove the cache in order to avoid the resubmissions of tasks whose DB informations could have been not completely consistent. 
It was just to avoid further troubles, you know the cure should not kill more than the desease. 

And I know what I'm talking about :P
